### PR TITLE
Dev s100/feature/185617 configure and consume sap endpoint for s 100 products

### PR DESCRIPTION
**Pull request by pratibh14837**
Hi,

Dependency :- no dependency

This PR contains code changes, unit tests to configure and consume sap endpoint for s 100 products.
Please review.

Regards!

### CODE REVIEW

**Review by jagannathsankpal — APPROVED** [review]
LGTM

File: src/UKHO.ERPFacade.API/Handlers/S100EventHandler.cs [hunk 1]
```diff
@@ -19,13 +23,22 @@ public class S100EventHandler : IEventHandler
         private readonly IAzureTableReaderWriter _azureTableReaderWriter;
         private readonly IAzureBlobReaderWriter _azureBlobReaderWriter;
         private readonly IBaseXmlTransformer _baseXmlTransformer;
+        private readonly ISapClient _sapClient;
+        private readonly IOptions<SapConfiguration> _sapConfig;
 
-        public S100EventHandler(ILogger<S100EventHandler> logger, IAzureTableReaderWriter azureTableReaderWriter, IAzureBlobReaderWriter azureBlobReaderWriter, [FromKeyedServices("S100XmlTransformer")] IBaseXmlTransformer baseXmlTransformer)
+        public S100EventHandler(ILogger<S100EventHandler> logger,
+            IAzureTableReaderWriter azureTableReaderWriter,
+            IAzureBlobReaderWriter azureBlobReaderWriter,
```
Comment (Vishal14583) [inline src/UKHO.ERPFacade.API/Handlers/S100EventHandler.cs:31]: Please correct the indentation. Refer S57EventHandler

File: src/UKHO.ERPFacade.Common/Logging/EventIds.cs [hunk 2]
```diff
@@ -477,7 +477,17 @@ public enum EventIds
         /// <summary>
         /// 940095 - Shared API Key Configuration is missing.
         /// </summary>
-        SharedApiKeyConfigurationMissing = 940095
+        SharedApiKeyConfigurationMissing = 940095,
+
+        /// <summary>
+        /// 940096 - S100 Request to SAP failed.
```
Comment (Vishal14583) [inline src/UKHO.ERPFacade.Common/Logging/EventIds.cs:483]:
```suggestion
        /// 940096 - S-100 Request to SAP failed.
```

File: src/UKHO.ERPFacade.Common/Logging/EventIds.cs [hunk 3]
```diff
@@ -477,7 +477,17 @@ public enum EventIds
...
+        /// </summary>
+        S100RequestToSapFailedException = 940096,
+
+        /// <summary>
+        /// 940097 - S100 data content sent to SAP.
```
Comment (Vishal14583) [inline src/UKHO.ERPFacade.Common/Logging/EventIds.cs:488]:
```suggestion
        /// 940097 - S-100 data content sent to SAP.
```

Comment (Vishal14583) [reply src/UKHO.ERPFacade.Common/Logging/EventIds.cs:488]: Please update the "S-100" across the project as per given naming convention wherever applicable.

File: tests/UKHO.ERPFacade.API.UnitTests/Handlers/S100EventHandlerTests.cs [hunk 4]
```diff
@@ -67,6 +81,34 @@ public void WhenValidS100EventHandlerEventReceived_ThenProcessingS100Event()
...
+            A.CallTo(_fakeLogger).Where(call => call.Method.Name == "Log"
+                                                && call.GetArgument<LogLevel>(0) == LogLevel.Information
+                                                && call.GetArgument<EventId>(1) == EventIds.S100EventUpdateSentToSap.ToEventId()
+                                                && call.GetArgument<IEnumerable<KeyValuePair<string, object>>>(2)!.ToDictionary(c => c.Key, c => c.Value)["{OriginalFormat}"].ToString() == "S-100 data content has been sent to SAP successfully.").MustHaveHappenedOnceExactly();
+
+            A.CallTo(() => _fakeAzureTableReaderWriter.UpdateEntityAsync(A<string>.Ignored, A<string>.Ignored, A<KeyValuePair<string, DateTime>[]>.Ignored)).MustHaveHappened();
+        }
+
+        [Test]
+        public void WhenS100EventHandler_NotProcessedTheEvent()
```
Comment (Vishal14583) [inline tests/UKHO.ERPFacade.API.UnitTests/Handlers/S100EventHandlerTests.cs:98]:
```suggestion
        public void WhenSapRespondsWith401Unauthorized_ThenS100EventHandlerRaiseAnException()
```

Comment (Vishal14583) [reply tests/UKHO.ERPFacade.API.UnitTests/Handlers/S100EventHandlerTests.cs:98]: Unit test case names should in form of "When<>_Then. It should include following.

- The name of the method being tested.
- The scenario under which it's being tested.
- The expected behavior when the scenario is invoked.

**Review by Vishal14583 — CHANGES_REQUESTED** [review]
Few review comments

File: src/UKHO.ERPFacade.Common/Logging/EventIds.cs (hunk 2 above)
Comment (pratibh14837) [reply src/UKHO.ERPFacade.Common/Logging/EventIds.cs]: resolved

File: src/UKHO.ERPFacade.Common/Logging/EventIds.cs (hunk 3 above)
Comment (pratibh14837) [reply src/UKHO.ERPFacade.Common/Logging/EventIds.cs]: Updated the "S-100" in EventIds.cs

File: tests/UKHO.ERPFacade.API.UnitTests/Handlers/S100EventHandlerTests.cs (hunk 4 above)
Comment (pratibh14837) [reply tests/UKHO.ERPFacade.API.UnitTests/Handlers/S100EventHandlerTests.cs]: Renamed Unit Test Name as per the above suggestion

**Review by Vishal14583 — APPROVED** [review]
LGTM

**Review by gaurav15233 — APPROVED** [review]
LGTM

**Review by rushdynajath — APPROVED** [review]
looks good

**Pull request merged**